### PR TITLE
[refactoring] Deprecate methods with correspondence in Guava

### PR DESCRIPTION
**Pull request by kthoms**
Also refactored methods to use Guava

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java [hunk 1]
```diff
@@ -139,64 +149,30 @@ public static String readFileIntoString(String filename) {
...
-			byte[] buffer = new byte[2048];
-			int bytesRead = 0;
-			StringBuffer b = new StringBuffer();
-			do {
-				bytesRead = source.read(buffer);
-				if (bytesRead != -1)
-					b.append(new String(buffer, 0, bytesRead));
-			} while (bytesRead != -1);
-			return b.toString();
+			return new String(ByteStreams.toByteArray(inputStream));
```
Comment (szarnekow) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java]: No longer buffered

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java [hunk 2]
```diff
@@ -139,64 +149,30 @@ public static String readFileIntoString(String filename) {
...
-			byte[] data = new byte[16384];
-	
-				while ((nRead = inputStream.read(data, 0, data.length)) != -1) {
-				  buffer.write(data, 0, nRead);
-				}
-	
-			buffer.flush();
-	
-			return buffer.toByteArray();
+			return ByteStreams.toByteArray(inputStream);
```
Comment (szarnekow) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java]: Please add buffering to the inputStream if it is not yet buffered

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java [hunk 3]
```diff
@@ -139,64 +149,30 @@ public static String readFileIntoString(String filename) {
 	
 	/**
 	 * @since 2.3
+	 * @deprecated Use com.google.common.io.ByteStreams.toByteArray(InputStream) instead
```
Comment (szarnekow) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java]: The deprecation comment is not valid if we do the additional buffering

**Comment by kthoms** [comment]
Thanks for these comments. Will change.

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java (hunk 2 above)
Comment (kthoms) [reply org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java]: toByteArray calls copy(in,out), which does buffering
com.google.common.io.ByteStreams.copy(InputStream, OutputStream)

Am I missing something?

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java (hunk 1 above)
Comment (kthoms) [reply org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java]: is buffered by toByteArray/copy, right?

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java (hunk 2 above)
Comment (szarnekow) [reply org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java]: I suspect that a buffered input stream with a buffer size of 16k is the better choice than a custom buffering in ByteStreams.copy with a buffer size of 4k. But that is only an assumption that would need to be verified.

Comment (kthoms) [reply org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java]: Actually there are no references to this method. Also an indicator to deprecate it?

Comment (szarnekow) [reply org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java]: Yes

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java (hunk 1 above)
Comment (szarnekow) [reply org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java]: true

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java (hunk 3 above)
Comment (szarnekow) [reply org.eclipse.xtext.util/src/org/eclipse/xtext/util/Files.java]: My comment is obsolete.

**Comment by cdietrich** [comment]
@szarnekow  what is the state of review?

**Comment by kthoms** [comment]
Thx, @szarnekow

**Pull request merged**